postupgrade/mergeGlobalConfigs.js

```diff
@@ -0,0 +1,174 @@
+   */
+  const parseCommentedOutProperty = (comment) => {
+    const propCommentRegex = /^\s?"([^"]+)":.*$/g
+    const val = comment.value
+    const match = [...val.matchAll(propCommentRegex)][0]
+    if (!match || match.length !== 2) {
+      throw new Error(`Could not parse global config commented out property: \`${val}\``)
+    }
+    return {
+      type: 'CommentedOutProperty',
+      key: match[1],
+      comment: {
+        type: comment.type,
+        value: comment.value,
+        inline: comment.inline
+      }
+    }
+  }
+
+  const handleComment = c => {
+    if (c.inline) return;
+    tokens.push(parseCommentedOutProperty(c));
+  }
+
+  Object.keys(commentJsonObject).forEach(key => {
+    const beforeSymbol = Symbol.for(`before:${key}`)
+    const afterSymbol = Symbol.for(`after:${key}`)
+    const beforeComments = commentJsonObject[beforeSymbol] || []
+    const afterComments = commentJsonObject[afterSymbol] || []
+    beforeComments.forEach(handleComment)
+    const inlineComment = afterComments.find(c => c.inline);
+    tokens.push({
+      type: 'Property',
+      key,
+      value: commentJsonObject[key],
+      inlineComment: inlineComment ? {
+        type: inlineComment.type,
+        value: inlineComment.value,
+        inline: inlineComment.inline
+      } : null
+    })
+    afterComments.forEach(handleComment)
+  })
+
+  return tokens;
+}
+exports.tokenize = tokenize;
+
+/**
+ * Transforms an array of {@link GlobalConfigToken} into an equivalent {@link CommentJSONObject}
+ * 
+ * @param {GlobalConfigToken[]} tokens 
+ * @returns {CommentJSONObject}
+ */
+function transformToCommentJsonObject(tokens) {
+  const commentJsonObject = {}
+  let previousPropertyKey = null;
+  tokens.forEach(token => {
+    switch (token.type) {
+      case 'Property':
+        commentJsonObject[token.key] = token.value
+        previousPropertyKey = token.key
+        if (token.inlineComment) {
+          const symbol = Symbol.for(`after:${token.key}`);
+          commentJsonObject[symbol] = (commentJsonObject[symbol] || []).concat([token.inlineComment])
+        }
+        break;
+      case 'CommentedOutProperty':
+        const symbol = previousPropertyKey ? Symbol.for(`after:${previousPropertyKey}`) : Symbol.for('before');
+        commentJsonObject[symbol] = (commentJsonObject[symbol] || []).concat([token.comment])
+        break;
+      default:
+        throw new Error(`Unknown token type "${token.type}"`);
+    }
+  });
+  return commentJsonObject;
+}
+exports.transformToCommentJsonObject = transformToCommentJsonObject;
+
+/**
+ * The logic for merging two arrays of {@link GlobalConfigToken}.
+ * 
+ * @param {GlobalConfigToken[]} original 
+ * @param {GlobalConfigToken[]} incoming 
+ * @returns 
+ */
+ function mergeTokens(original, incoming) {
+  const mergedTokens = []
+  const originalCopy = [...original]
+
+  incoming.forEach(incomingToken => {
+    const i = originalCopy.findIndex(t => t.key === incomingToken.key)
```

Comment:
Would it make sense to store the tokens in a Map instead of an array? That could save us this `findIndex` call here. Both `incoming` and `original` would have tokens by key.

Reply:
The order of the tokens is important, so I opted for an array. ES6 Maps preserve order by insertion order but I was worried about that being a point of failure in the future, say if we need to refactor this code and somebody isn't careful about insertion order.